### ISSUE TITLE
Wrong invalid-argument-type for list comprehension

### ISSUE THREAD

**Issue by wardVD** [astral-sh/ty#2355]
### Summary

We get an [invalid-argument-type](https://ty.dev/rules#invalid-argument-type) for the following code:

```python
from typing import Literal

def test(test_string: Literal["a", "b"]) -> str:
    return test_string

[test(string) for string in ["a", "b"]]
```

The `string` variable returns

```
Argument to function `test` is incorrect: Expected `Literal["a", "b"]`, found `Unknown | str` ty[invalid-argument-type](https://ty.dev/rules#invalid-argument-type)
```

From the code it should be clear that the parameters can only be `"a"` or `"b"` so I believe this is a bug.

### Version

uv 0.9.21 (Homebrew 2025-12-30)

**Comment by ibraheemdev** [comment]
Thanks for the report! This is a known bug, we don't yet support bidirectional inference with list comprehensions, and so aren't able to avoid literal promotion.
